design-docs/De-inline nulls and errors.md

```diff
@@ -0,0 +1,147 @@
+# Proposal: de-inline nulls and errors in normalization
+
+## Context: how it works today
+
+When normalizing responses currently only JSON objects are de-inlined and stored in their own records.
+The value of the field in the parent record is then a reference (CacheKey) to that child record.
+
+Other types of values (primitive types, nulls, errors, and arrays) are kept inline in their parent record or array.
```

Comment:
Actually, not true, JSON objects are kept inlined, if they are scalars. I'll amend.